### PR TITLE
Confirmations: Empty SiteURL and Null TagURL in modal

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
There are two issues that show up on the approve confirmation modal when you deeplink to the app. The first is a null tagURL which is well documented [here](https://github.com/MetaMask/mobile-planning/issues/883) and the second is an undefined Site URL value well documented [here](https://github.com/MetaMask/metamask-mobile/issues/5570). Following the conversations on the issues and [here](https://github.com/MetaMask/mobile-planning/issues/883#issuecomment-1514748749), [here](https://github.com/MetaMask/metamask-mobile/issues/5570#issuecomment-1402031558), this PR includes a fix that hides both the tagurl and siteurl when not available.

**Screenshots/Recordings**

https://user-images.githubusercontent.com/29962968/234585202-0fc35576-6d2f-47ce-a729-b5d83335d1d2.MP4

**Issue**

Progresses #5570 and https://github.com/MetaMask/mobile-planning/issues/883

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Review by jpuri — APPROVED** [review]
Hey @blackdevelopa , changes look good. Will be nice to have some test coverage also.

File: app/components/UI/TransactionReview/TransactionReviewDetailsCard/index.js [hunk 1]
```diff
@@ -135,12 +135,14 @@ export default class TransactionReviewDetailsCard extends Component {
           title={strings('spend_limit_edition.transaction_details')}
         />
         <View style={styles.transactionDetails}>
-          <View style={styles.transactionDetailsRow}>
-            <Text style={styles.transactionDetailsTextLeft}>
-              {strings('spend_limit_edition.site_url')}
-            </Text>
-            <Text style={styles.transactionDetailsTextRight}>{host}</Text>
-          </View>
+          {host ? (
```
Comment (segun) [inline app/components/UI/TransactionReview/TransactionReviewDetailsCard/index.js:138]: Same here, we should add some tests to check this condition

File: app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.tsx [hunk 2]
```diff
@@ -145,11 +145,13 @@ const ApproveTransactionHeader = ({
 
   return (
     <View style={styles.transactionHeader}>
-      <TagUrl
-        imageSource={{ uri: favIconUrl }}
-        label={domainTitle}
-        style={styles.tagUrl}
-      />
+      {origin ? (
```
[the remark this answers is not in the document]
Comment (blackdevelopa) [reply app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.tsx:135]: Updated. TY [d859ddb](https://github.com/MetaMask/metamask-mobile/pull/6274/commits/d859ddbd1769bcf6aaaea45adb81a124c4f7fd97)

File: app/components/UI/TransactionReview/TransactionReviewDetailsCard/index.js (hunk 1 above)
Comment (blackdevelopa) [reply app/components/UI/TransactionReview/TransactionReviewDetailsCard/index.js:138]: Updated. [d859ddb](https://github.com/MetaMask/metamask-mobile/pull/6274/commits/d859ddbd1769bcf6aaaea45adb81a124c4f7fd97)

**Review by segun — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Pull request merged**